src/utils/channel_utils.js

```diff
@@ -24,7 +32,7 @@ const channelTypeOrder = {
  *  favoriteChannels: [...]
  * }
  */
-export function buildDisplayableChannelList(usersState, allChannels, myMembers, config, myPreferences, teammateNameDisplay, lastPosts) {
+export function buildDisplayableChannelList(usersState: UsersState, allChannels: Array<Channel>, myMembers: {[string]: ChannelMembership}, config: Object, myPreferences: {[string]: PreferenceType}, teammateNameDisplay: string, lastPosts: {[string]: Post}) {
```

Comment:
Yes, probably would be super cool to be able to generate the client types from anywhere. But, probably a lot of the current existing types are already defined as part of the GlobalState.

Reply:
I actually meant the client version of config.json, since it's just a `map[string]string` on the server side ([link](https://github.com/mattermost/mattermost-server/blob/master/utils/config.go#L427)). Here, we're using a plain Object which works, but it would be nice to have a proper type with all the field names, even if they are just all strings